### ISSUE TITLE
QBtn type button => submit 

### ISSUE THREAD

**Issue by florin-rotaru** [quasarframework/quasar#11794]
**QBtn of type button should or shouldn't trigger the QForm submit action ?**

Seems that with version 2.4.1 (possible even earlier ones after 2.3.4) by clicking a QBtn of type button will trigger the QForm submit action.

**Codepen**
[https://codepen.io/florin-rotaru/pen/yLzparG](https://codepen.io/florin-rotaru/pen/yLzparG)

**Platform (please complete the following information):**
Quasar Version:             2.4.1
@quasar/app Version:  3.2.6

Quasar mode:
  - [x] SPA

Additonal Info:
- [X] Typescript
- [X] Vuex
- [X] Standard (eslint)
  
Tested on:
  - [X] SPA

OS:
Node: 16.13.1
NPM: 8.3.0
Yarn: 1.22.17
Browsers: Edge / Firefox

**Comment by rstoenescu** [comment]
Hi,

Whatever the browser behavior is with native `<button>`s so is Quasar's. At its base, QBtn's type prop is set as attribute to the native rendered `<button>`. The only comment I have with your codepen is that QBtn has implicit type="button", so specifying it as "button" is redundant.

**Comment by rstoenescu** [comment]
And what I mean by "native behavior" is:

```
<form @submit.prevent="onFormSubmit">
      <div class="q-ma-md">
        <input v-model="gigi">
        <button
          color="red"
          @click="notify">
          Cancel
        </button>
        <button
          type="submit"
          color="primary"
          @click="notify">
          Save
        </button>
      </div>
    </form>
```

**Comment by rstoenescu** [comment]
Hmm, missed one important thing in regards to the native behavior. That if we render the redundant `type="button"` then the browser actually hinders the submit behavior. So Quasar should actually add the type=button attribute.

**Comment by rstoenescu** [comment]
Enhancement will be available in Quasar v2.4.2.
